### ISSUE TITLE
How to display Message, Notifications... using the composition API?

### ISSUE THREAD

**Issue by vowar** [element-plus/element-plus#436]
Hello! I'm new to the Vue and currently, I'm working with Vue3( composition API ), Vuex4, ElementPlus. And I had a case where I needed to display an error message in a Vuex action.

The first thing that I tried was the **Local import** of the `Message` as mentioned in the https://element-plus.org/#/en-US/component/message

```js
import { Message } from 'element-plus';

const authModule: Module<State, RootState> = {
  ...
  actions: {
    someAction() {
      try {
        ...
      } catch(error) {
        ...
        Message.error({ message: error.message })
      }
    }
  }
}

```
But if I try to import it this way, the `Message` is always `undefined`

After, I tried to watch the `error` inside the `setup` function and wanted to use the `$message`, but soon I realized that I can't do this too because I don't have access to `this`.

```js
setup() {
  ...
  watch(error, () => {
    ... // can't invoke `this.$message({ message: error })`
  })
  return { ... }
},
```

How can I call the `Message` other than calling the `this.$message`? Will it be stuff like `provide()/inject()`? Or just the **Local import** should work?

**Comment by JeremyWuuuuu** [comment]
Local `import` should work, you can refer to `notification/src/notify` file, that might be a mistake for me to not export the notify instance in the `index` file. You can try if it works, if not, I might need to make a PR for exposing the instance then.

Sorry for the inconvenience.

**Comment by vowar** [comment]
I've tried to `import Message from 'element-plus/packages/message/src/message'` but got the following error

```
 ERROR  Failed to compile with 3 errors                                                                                     11:47:12 AM

These dependencies were not found:

* @element-plus/utils/isServer in ./node_modules/element-plus/packages/message/src/message.ts
* @element-plus/utils/popup-manager in ./node_modules/element-plus/packages/message/src/message.ts
* @element-plus/utils/util in ./node_modules/element-plus/packages/message/src/message.ts
```

Version  `1.0.1-alpha.6`

Was it the right import?

**Comment by JeremyWuuuuu** [comment]
@vowar  Seems we need to export the entry later, I'll make a fix maybe later today.

**Comment by vowar** [comment]
@JeremyWuuuuu  Thanks 👍

**Comment by uranius666** [comment]
Not working yet)

**Comment by toramanlis** [comment]
> I've tried to `import Message from 'element-plus/packages/message/src/message'` but got the following error
> 
> ```
>  ERROR  Failed to compile with 3 errors                                                                                     11:47:12 AM
> 
> These dependencies were not found:
> 
> * @element-plus/utils/isServer in ./node_modules/element-plus/packages/message/src/message.ts
> * @element-plus/utils/popup-manager in ./node_modules/element-plus/packages/message/src/message.ts
> * @element-plus/utils/util in ./node_modules/element-plus/packages/message/src/message.ts
> ```
> 
> Version `1.0.1-alpha.6`
> 
> Was it the right import?

You can import `element-plus/lib/message`

**Comment by ccfish86** [comment]
```
import { Message } from "element-plus/lib/message";
import { MessageBox } from "element-plus/lib/message-box";
```
like this. 
and full example 'https://github.com/ccfish86/demo-vue3/blob/master/element-plus/src/api/request.ts#L9'

**Comment by zazzaz** [comment]
we have fixed this in #630 and will be available in the next release

**Comment by zazzaz** [comment]
fixed in 1.0.1-alpha.18

Now you can use 
```
import { ElMessageBox, ElMessage } from "element-plus"
```

Docs will be updated shortly

**Comment by loloof64** [comment]
I managed for notification with 
```javascript
import { ElNotification } from "element-plus";
...
ElNotification({
   title: "Error",
   type: "error",
   message: "Could not set engine path !",
});
```

**Comment by AlekseyMalakhov** [comment]
This syntax works as well
```
ElNotification.error({
     title: 'Error',
     message: 'This is an error message'
});
```

**Comment by soichih** [comment]
Does `ElNotification` only work on composition API? I am not using composition API, and when I call ELNotification, I see the dom elements with el-notification created under `body`, but it's displayed beloved the bottom edge of the page so they are not visible. I feel like I am missing some stylesheet for ElNotification, but everything else from element-plus works.. so I am not sure why it doesn't work with just ElNotification.

![Screenshot from 2021-11-04 15-15-07](https://user-images.githubusercontent.com/923896/140405279-4a2e272a-ebda-4537-a0ee-68c01fd9cd96.png)

**Comment by sxzz** [comment]
@soichih You should import style.

```js
import 'element-plus/dist/index.css'
```

Reference: https://element-plus.org/en-US/guide/quickstart.html

**Comment by soichih** [comment]
Ah, thank you! I was using the vite resolver (`ElementPlusResolver`) and I thought it took care of it. Thanks!

**Comment by sxzz** [comment]
@soichih `unplugin-vue-components` will only resolve components, not for API.

**Comment by soichih** [comment]
@sxzz It resolves components and their CSS, right? What's odd is that, `import 'element-plus/dist/index.css'` will also import the CSS for components.. so by doing that, it kind of defeats the purpose of `ElementPlusResolver`.. I only want to import components and their CSS that are used by the App. Is there CSS rule that I can import that only includes CSS used by the API?

**Comment by sxzz** [comment]
Yeah. You can import `element-plus/es/components/el-xxx/style/css`

**Comment by soichih** [comment]
Hum.. so I have to load separate CSS used by each API component?

So for ElNotification, I am guessing it's under "notification" like this?

```
import { ElNotification } from 'element-plus'
import 'element-plus/es/components/notification/style/css'
```

It's a bit awkward.. but I guess I can live with it. Thanks!

**Comment by sxzz** [comment]
We will improve it in the future!

**Comment by sxzz** [comment]
Also, only message, message-box, notification, loading should be imported by yourself. The others are components that will import style automatically.

**Comment by fanhefeng** [comment]
> Also, only message, message-box, notification, loading should be imported by yourself. The others are components that will import style automatically.
![1](https://user-images.githubusercontent.com/28811207/141079551-9a9ee4a6-7c67-414c-870c-1a44b37a2069.png)
finally~

**Comment by soichih** [comment]
I just noticed that none of the element UI's icons are displayed either. I had to install a separate module "npm install @element-plus/icons" (why!?) Then I have to manually import them on all the components that I am using those icons (Why won't the ElementPlusResolve work?). After that, I was able to display the icon by itself, 

```
<Remove/>
```

![image](https://user-images.githubusercontent.com/923896/144124195-9506bac5-df5f-4fa5-beb6-701b5d4d60ba.png)

Yay!

But if I put it inside other element UI components, nothings gets displayed.. like
```
<el-button type="danger" @click="remove(key)" size="mini"><Remove/></el-button>  
``` 

Results in 

![image](https://user-images.githubusercontent.com/923896/144124092-62c198c5-6226-4afa-9685-223328e73344.png)

Why?
